### DISCUSSION TITLE
training.tsv Column 3?

### DISCUSSION THREAD

**Discussion by VikR0001** [explosion/spaCy#9092] [Help: Coding & Implementations] [answered]
I cloned `textcat_goemotions` from the tutorials, installed the assets, and ran `python -m spacy project run all`. It ran successfully and created the binary wheel `en_textcat_goemotions-0.0.1-py3-none-any.whl`.  

Now I would like to drop in my actual training data. Looking at the file `assets/train.tsv`, I see 3 columns:

<img width="639" alt="Screen Shot 2021-08-30 at 2 07 38 PM" src="https://user-images.githubusercontent.com/10912958/131405785-cd2119bb-abf1-4945-a3b9-8316c7ff2164.png">

I'm pretty sure column 2 is a 0-based index into the list of emotions in `assets/categories.txt`. But I'm not sure yet what the codes are in column 3.

What are those codes in column 3 - and, do I need to generate them in order to use spacy to train a model with my data?

**Comment by polm** [comment, marked as answer]
If you look at the [preprocessing script](https://github.com/explosion/projects/blob/a038a5bb75f0aa53948ba810fbf4dd795550e967/tutorials/textcat_goemotions/scripts/convert_corpus.py#L15) you'll see that that field is labelled "annotator". It's not used in creating the training data (see `convert_record`), so you don't need it.

In general, even if the example projects don't have a full README for the source data, the preprocessing scripts should make it clear what the various fields are and how they're used.

**Reply by VikR0001** [reply]
I'm clicking "Mark as answer" but it isn't being marked for some reason. :)
